language/evm/move-to-yul/src/evm_transformation.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Performs transformations specific for compiling stackless bytecode to
+//! EVM bytecode. Right now it only contains conversions of U256 native functions.
+
+use ethnum::U256;
+use move_model::model::FunctionEnv;
+use move_stackless_bytecode::{
+    function_target::FunctionData,
+    function_target_pipeline::{FunctionTargetProcessor, FunctionTargetsHolder},
+    stackless_bytecode::{Bytecode, Constant, Operation},
+};
+
+const CAST_TO_U256_FUNCTION_NAME: &str = "u256_from_words";
+
+pub struct EvmTransformationProcessor {}
+
+impl EvmTransformationProcessor {
+    pub fn new() -> Box<Self> {
+        Box::new(EvmTransformationProcessor {})
+    }
+}
+
+impl FunctionTargetProcessor for EvmTransformationProcessor {
+    fn process(
+        &self,
+        _targets: &mut FunctionTargetsHolder,
+        func_env: &FunctionEnv<'_>,
+        mut data: FunctionData,
+    ) -> FunctionData {
+        if func_env.is_native() || func_env.is_intrinsic() {
+            return data;
+        }
+        for code_offset in 0..data.code.len() {
+            let bytecode = &data.code[code_offset];
+            if let Bytecode::Call(
+                attr_id,
+                dst,
+                Operation::Function(module_id, fun_id, _),
+                src,
+                abort_condition,
+            ) = bytecode
+            {
+                if crate::generator::is_evm_arith_module(
+                    &func_env.module_env.env.get_module(*module_id),
+                ) {
+                    let fun_name = &*func_env
+                        .module_env
+                        .env
+                        .get_module(*module_id)
+                        .get_function(*fun_id)
+                        .get_simple_name_string();
```

Comment:
Should this be better the qualified name you check for?

Reply:
Qualified name meaning something like "0x1::U256::add"? Since earlier we already checked on line 45 the module identified by module_id is an evm_arith module, I think it's sufficient to check simple name?

Reply:
Ah ok.